plugins/modules/vmware_host_access_manager.py

```diff
@@ -0,0 +1,362 @@
+            self.module.fail_json(msg="Failed to update permission for '%s' due to user"
+                                      " not found : %s" % (self.principal, to_native(user_not_found.msg)))
+        except Exception as generic_exc:
+            self.module.fail_json(msg="Failed to add access: %s" % (to_native(generic_exc)))
+
+    def get_lockdown_mode(self, access_mgr):
+        """
+        Get the lockdown mode from ESXi
+
+        Returns: Lockdown mode as string
+        """
+        lockdown_mode = access_mgr.lockdownMode
+        return {v: k for k, v in self.LOCKDOWN_MODE_MAP.items()}.get(lockdown_mode)
+
+    def execute_access_controll_entry(self, host, access_mgr):
+        access_mode_current = self.get_access_controll_entry(access_mgr)
+
+        if self.state == "present":
+            if access_mode_current != self.access_mode:
+                self.result['changed'] = True
+                if self.module.check_mode:
+                    self.result['msg'].append("The permission '%s' will be granted to '%s' on '%s'." %
+                                              (self.access_mode, self.principal, host.name))
+                else:
+                    self.result['msg'].append("Granted permission '%s' to '%s' on '%s'" %
+                                              (self.access_mode, self.principal, host.name))
+                    self.update_access_controll_entry(access_mgr, self.access_mode)
+
+                self.result['result'][host.name]['access'] = {self.principal: dict(group=self.group, access=self.access_mode)}
+                if self.module._diff:
+                    self.result['diff']['before'][host.name]['access'] = [dict(principal=self.principal, group=self.group, access=access_mode_current)] if access_mode_current else []
+                    self.result['diff']['after'][host.name]['access'] = [dict(principal=self.principal, group=self.group, access=self.access_mode)]
+        elif self.state == "absent" and access_mode_current:
+            self.result['changed'] = True
+            if self.module.check_mode:
+                self.result['msg'].append("The permission '%s' will be revoked from '%s' on '%s'." %
+                                          (access_mode_current, self.principal, host.name))
+            else:
+                self.result['msg'].append("Revoked permission '%s' from '%s' on '%s'" %
+                                          (access_mode_current, self.principal, host.name))
+                self.update_access_controll_entry(access_mgr, None)
+
+            if self.module._diff:
+                self.result['diff']['before'][host.name]['access'] = [dict(principal=self.principal, group=self.group, access=access_mode_current)]
+                self.result['diff']['after'][host.name]['access'] = []
+
+    def execute_lockdown_exceptions(self, host, access_mgr):
+        exceptions = [s for s in access_mgr.QueryLockdownExceptions()]
```

Comment:
Pylint unnecessary-comprehension: Unnecessary use of a comprehension, use list(access_mgr.QueryLockdownExceptions()) instead.